UIforETW/UIforETWDlg.cpp

```diff
@@ -1646,7 +1646,7 @@ void CUIforETWDlg::PreprocessTrace(const std::wstring& traceFilename)
 		child.Run(bShowCommands_, L"xperf.exe" + args);
 		std::wstring output = child.GetOutput();
 		std::map<std::wstring, std::vector<DWORD>> pidsByType;
-		for (auto& line : split(output, '\n'))
+		for (const auto line : split(output, '\n'))

```

Comment:
> I think you're confusing the result of split() and the 'line' variable. 

That's exactly what I was doing. I still find it a bit surprising, but that's fair enough.


> I also like that the line of code ends up looking pythonic when we iterate over the return value directly. When C++11 lets code look more like Python I think that is a good thing.

I agree entirely. Well, *almost* entirely. I really love `Python`, but it has some very strange scoping rules. [Loop variables leak into the enclosing scopes](http://eli.thegreenplace.net/2015/the-scope-of-index-variables-in-pythons-for-loops), which is *crazy* from a `C` state of mind.